client/my-sites/themes/themes-banner/random-themes-banner.jsx

```diff
@@ -0,0 +1,43 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { sample } from 'lodash';
+
+const bannerCache = {};
+
+const isClient = () => {
+	return !! global.window;
+};
+
+class RandomThemesBanner extends PureComponent {
+	static propTypes = {
+		context: PropTypes.string.isRequired,
+		banners: PropTypes.array.isRequired,
+	};
+
+	constructor( props ) {
+		super( props );
+		const { context, banners } = props;
+		if ( ! bannerCache.hasOwnProperty( context ) ) {
+			bannerCache[ context ] = sample( banners );
+		}
```

Comment:
Hmm, can we drop the module-level banner cache? It kind of breaks encapsulation. Would be nice if we could just set the random banner at instance level.

Reply:
I'd also drop `context` TBH, because, you've guessed it, YAGNI 😉